#! /usr/bin/env python

PACKAGE="vlm_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        name        type     level    description       default  min  max
gen.add("queries",   str_t, 0, "Queries for VLM model. You can provide batch queries with ;", "What does this image describe?;")
gen.add("do_sample", bool_t, 0, "Whether or not to use sampling. If True, generate answers randomly", True)
gen.add("top_k",     int_t,  0, "The number of highest sentence probabilities to keep in top-k-filtering.", 5, 1, 10)
gen.add("max_length",int_t,  0, "The maximum length of the sequence to be generated.", 512, 1, 1024)
exit(gen.generate(PACKAGE, "vlm_ros", "VLM"))
